simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -25,6 +25,13 @@
 
 #include <immintrin.h>
 
+// FIXME_HIP ROCm 5.6 can't compile with the intrinsic used here.
+#if defined(__HIPCC__) &&       \
+    ((HIP_VERSION_MAJOR > 5) || \
+     ((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR >= 6)))
```

Comment:
Why don't you limit the workaround to ROCm 5.6? There is the same problem with 5.7?

Reply:
I am being conservative. We can restrict the workaround if we know that 5.7 works.

Reply:
If you haven't check 5.7, then we should try it before merging this.

Reply:
Please do it yourself then. I don't have access to 5.7 on a machine with AMD GPUs.

Reply:
Just trying with `Serial`, I am still seeing the error with 5.7:
```
cd /app/kokkos/build_hip/simd/unit_tests && /opt/rocm/bin/hipcc  -DKOKKOS_DEPENDENCE -I/app/kokkos/simd/unit_tests/include -I/app/kokkos/tpls/gtest -I/app/kokkos/build_hip -I/app/kokkos/build_hip/core/src -I/app/kokkos/core/src -I/app/kokkos/core/src/../../tpls/desul/include -I/app/kokkos/build_hip/containers/src -I/app/kokkos/containers/src -I/app/kokkos/build_hip/algorithms/src -I/app/kokkos/algorithms/src -I/app/kokkos/build_hip/simd/src -I/app/kokkos/simd/src  -O2 -g -DNDEBUG   -pthread -march=znver2 -mtune=znver2 -std=gnu++17 -o CMakeFiles/Kokkos_UnitTest_SIMD.dir/TestSIMD.cpp.o -c /app/kokkos/simd/unit_tests/TestSIMD.cpp
fatal error: error in backend: Instruction Combining seems stuck in an infinite loop after 1000 iterations.
clang++: error: clang frontend command failed with exit code 70 (use -v to see invocation)
AMD clang version 17.0.0 (https://github.com/RadeonOpenCompute/llvm-project roc-5.7.0 23352 d1e13c532a947d0cbfc94759c00dcf152294aa13)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/rocm-5.7.0/llvm/bin
clang++: note: diagnostic msg: Error generating preprocessed source(s).
```

Reply:
Let's change the condition to only check for ROCm 5.6 and 5.7. I've talked with AMD and it's easier for the compiler team to fix the issue if they can just pull Kokkos without having to mess with our code to reproduce the error.